felix/calc/l3_route_resolver.go

```diff
@@ -848,11 +851,11 @@ func (c *L3RouteResolver) flush() {
 				}
 			}
 		}
-		rt.SameSubnet = poolAllowsCrossSubnet && c.nodeInOurSubnet(rt.DstNodeName)
+		rt.SameSubnet = poolAllowsCrossSubnet && c.nodeInOurSubnet(rt.DstNodeName, ipFamily)
 
 		if rt.Dst != emptyV4Addr.AsCIDR().String() && rt.Dst != emptyV6Addr.AsCIDR().String() {
 			// Skip sending a route for an empty CIDR
-			logrus.WithField("route", rt).Debug("Sending route")
+			logrus.WithField("route", rt).Debug("Sending route SameSubnet=%v")
```

Comment:
```suggestion
			logrus.WithFields(logrus.Fields{
			    "route":                  rt,
			    "rt.SameSubnet": rt.SameSubnet,
			}).Debug("Sending route")
```

either that or remove "SameSubnet=%v" from the string?